pkg/storage/unified/resource/storage_backend_test.go

```diff
@@ -0,0 +1,1030 @@
+				namespace       string
+				resourceVersion int64
+				value           []byte
+			}{
+				name:            iter.Name(),
+				namespace:       iter.Namespace(),
+				resourceVersion: iter.ResourceVersion(),
+				value:           iter.Value(),
+			})
+		}
+		return iter.Error()
+	})
+
+	require.NoError(t, err)
+	require.Greater(t, rv, int64(0))
+	require.Len(t, collectedItems, 2) // Only resources in "apps" group
+
+	// Verify the items contain expected data
+	names := make([]string, len(collectedItems))
+	for i, item := range collectedItems {
+		names[i] = item.name
+		require.Equal(t, "default", item.namespace)
+		require.Greater(t, item.resourceVersion, int64(0))
+		require.NotEmpty(t, item.value)
+	}
+	require.Contains(t, names, "resource-1")
+	require.Contains(t, names, "resource-2")
+}
+
+func TestKvStorageBackend_ListIterator_WithPagination(t *testing.T) {
+	backend := setupTestStorageBackend(t)
+	ctx := context.Background()
+
+	// Create multiple test resources
+	for i := 1; i <= 5; i++ {
+		testObj, err := createTestObjectWithName(fmt.Sprintf("resource-%d", i), "apps", fmt.Sprintf("data-%d", i))
+		require.NoError(t, err)
+
+		metaAccessor, err := utils.MetaAccessor(testObj)
+		require.NoError(t, err)
+
+		writeEvent := WriteEvent{
+			Type: resourcepb.WatchEvent_ADDED,
+			Key: &resourcepb.ResourceKey{
+				Namespace: "default",
+				Group:     "apps",
+				Resource:  "deployments",
+				Name:      fmt.Sprintf("resource-%d", i),
+			},
+			Value:      objectToJSONBytes(t, testObj),
+			Object:     metaAccessor,
+			PreviousRV: 0,
+		}
+
+		_, err = backend.WriteEvent(ctx, writeEvent)
+		require.NoError(t, err)
+	}
+
+	// First page with limit 2
+	listReq := &resourcepb.ListRequest{
+		Options: &resourcepb.ListOptions{
+			Key: &resourcepb.ResourceKey{
+				Namespace: "default",
+				Group:     "apps",
+				Resource:  "deployments",
+			},
+		},
+		Limit: 2,
+	}
+
+	var firstPageItems []string
+	var continueToken string
+
+	rv, err := backend.ListIterator(ctx, listReq, func(iter ListIterator) error {
+		count := 0
+		for iter.Next() {
+			if err := iter.Error(); err != nil {
+				return err
+			}
+			firstPageItems = append(firstPageItems, iter.Name())
+			count++
+			// Simulate pagination by getting continue token after limit items
+			if count >= int(listReq.Limit) {
+				continueToken = iter.ContinueToken()
+				break
+			}
+		}
+		return iter.Error()
+	})
+
+	require.NoError(t, err)
+	require.Greater(t, rv, int64(0))
+	require.Len(t, firstPageItems, 2)
+	require.NotEmpty(t, continueToken)
+
+	// Second page using continue token
+	listReq.NextPageToken = continueToken
+	var secondPageItems []string
+
+	_, err = backend.ListIterator(ctx, listReq, func(iter ListIterator) error {
+		for iter.Next() {
+			if err := iter.Error(); err != nil {
+				return err
+			}
+			secondPageItems = append(secondPageItems, iter.Name())
+		}
+		return iter.Error()
+	})
+
+	require.NoError(t, err)
+	require.Greater(t, len(secondPageItems), 0)
```

Comment:
Can we be more specific (there should be 2, right?)? We could also verify that we've received exact names that we expect.

Reply:
Hum. That's right !

Reply:
Ok. There is a bug to be fixed. Added a comment for now.